### ISSUE TITLE
Cannot 

### ISSUE THREAD

**Issue by kielczu** [RobHofmann/HomeAssistant-GreeClimateComponent#216]
I have the same problem with XE7A-24/HC. I have added two devices succesfully, but the third one is always unavailable.

Gree A/C version 2.14.2.
Wifi version v1.38.

These are my settings:

platform: gree
name: GREE-S1
host: 192.168.8.221
port: 7000
mac: '50:2C:C6:DF:73:21'
target_temp_step: 1
platform: gree
name: GREE-S3
host: 192.168.8.223
port: 7000
mac: 'C0:39:37:0F:E9:A3'
target_temp_step: 1
encryption_version: 2
platform: gree
name: GREE-S2
host: 192.168.8.222
port: 7000
mac: '94:24:B8:01:7A:44'
target_temp_step: 1
And the log for GREE-S3 is always:
2024-10-27 14:41:44.590 INFO (MainThread) [custom_components.gree.climate] Setting up Gree climate platform
2024-10-27 14:41:44.590 INFO (MainThread) [custom_components.gree.climate] Adding Gree climate device to hass
2024-10-27 14:41:44.590 INFO (MainThread) [custom_components.gree.climate] Initialize the GREE climate device
2024-10-27 14:41:44.590 INFO (MainThread) [custom_components.gree.climate] name(): GREE-S3
2024-10-27 14:41:44.590 INFO (MainThread) [custom_components.gree.climate] name(): GREE-S3
2024-10-27 14:41:44.591 INFO (MainThread) [custom_components.gree.climate] supported_features(): 425
2024-10-27 14:41:44.591 INFO (MainThread) [custom_components.gree.climate] temperature_unit(): ..C
2024-10-27 14:41:44.591 INFO (MainThread) [custom_components.gree.climate] hvac_modes(): [<HVACMode.AUTO: 'auto'>, <HVACMode.COOL: 'cool'>, <HVACMode.DRY: 'dry'>, <HVACMode.FAN_ONLY: 'fan_only'>, <HVACMode.HEAT: 'heat'>, <HVACMode.OFF: 'off'>]
2024-10-27 14:41:44.591 INFO (MainThread) [custom_components.gree.climate] min_temp(): 16
2024-10-27 14:41:44.591 INFO (MainThread) [custom_components.gree.climate] max_temp(): 30
2024-10-27 14:41:44.591 INFO (MainThread) [custom_components.gree.climate] target_temperature_step(): 1.0
2024-10-27 14:41:44.591 INFO (MainThread) [custom_components.gree.climate] fan_list(): ['Auto', 'Low', 'Medium-Low', 'Medium', 'Medium-High', 'High', 'Turbo', 'Quiet']
2024-10-27 14:41:44.591 INFO (MainThread) [custom_components.gree.climate] swing_modes(): ['Default', 'Swing in full range', 'Fixed in the upmost position', 'Fixed in the middle-up position', 'Fixed in the middle position', 'Fixed in the middle-
2024-10-27 14:41:44.592 INFO (MainThread) [custom_components.gree.climate] supported_features(): 425
2024-10-27 14:41:44.592 INFO (MainThread) [custom_components.gree.climate] Gree climate device added to hass()
2024-10-27 14:41:44.592 INFO (MainThread) [custom_components.gree.climate] update()
2024-10-27 14:41:44.592 INFO (MainThread) [custom_components.gree.climate] Retrieving HVAC encryption key
2024-10-27 14:41:44.593 INFO (MainThread) [custom_components.gree.climate] Fetching(192.168.8.223, 7000, 10, {"cid": "app","i": 1,"pack": "cF75QLiR1N9+XupJ+VKg4Z3xM9Wmdxq80T4WdlHY+7ej9mrs/0p7GUTdWxmNaBUt","t":"pack","tcid":"c039370fe9a3","uid":
2024-10-27 14:41:44.593 INFO (MainThread) [custom_components.gree.climate] 3
2024-10-27 14:41:54.604 INFO (MainThread) [custom_components.gree.climate] Error getting device encryption key!

Please assist.

**Comment by stenvix** [comment]
I think you should try to set encryption_version: 2 @kielczu @bugac

**Comment by kielczu** [comment]
I have spoken to gree suppor and i found out that there are different versions of the same XE7A-24/HC wifi module. The one which does not work (failed to get encryption key) apparently is a newer one that only uses bluetooth and does not have the legacy hotspot feature.

So it seems when we add the new module using bleetooth the encryption key is different.

Can anybody help with that? Perhaps the default encryption keys must be updated in the code?

**Comment by Dilergore** [comment]
> I have spoken to gree suppor and i found out that there are different versions of the same XE7A-24/HC wifi module. The one which does not work (failed to get encryption key) apparently is a newer one that only uses bluetooth and does not have the legacy hotspot feature.
> 
> So it seems when we add the new module using bleetooth the encryption key is different.
> 
> Can anybody help with that? Perhaps the default encryption keys must be updated in the code?

I have the same issue. There is no hotspot, I can connect it only using the app (the old netcat way is not working) and then in HA I'm getting "Error getting device encryption key".

**Comment by piotrgt** [comment]
> I have spoken to gree suppor and i found out that there are different versions of the same XE7A-24/HC wifi module. The one which does not work (failed to get encryption key) apparently is a newer one that only uses bluetooth and does not have the legacy hotspot feature.
> 
> So it seems when we add the new module using bleetooth the encryption key is different.
> 
> Can anybody help with that? Perhaps the default encryption keys must be updated in the code?



> I have spoken to gree suppor and i found out that there are different versions of the same XE7A-24/HC wifi module. The one which does not work (failed to get encryption key) apparently is a newer one that only uses bluetooth and does not have the legacy hotspot feature.
> 
> So it seems when we add the new module using bleetooth the encryption key is different.
> 
> Can anybody help with that? Perhaps the default encryption keys must be updated in the code?

Did you ever manage to resolve this issue? I just installed two AC units. One of them is split type and it connected straight away using this integration but other one, duct type AC using XE7A-24/HC controller does not connect at all. I tried a few times with different encryption key extraction methods but without luck.

**Comment by kielczu** [comment]
No i do not. I use the two that worked and did not integrate the third one, that uses the "new wifi module" that does not support WiFi hotspot and uses bluetooth only.

**Comment by RobHofmann** [comment]
Wait a minute. "new wifi module" that only supports Blutooth? If it's Blutooth only, it won't work with this module unfortunately.

**Comment by kielczu** [comment]
I mean that the new module can only use bluetooth to connect to the phone and establish wifi cconnection. Then it uses wifi for connectivity. The new module does not support wifi hotspot to pair with the gree app

**Comment by RobHofmann** [comment]
Ah right. Did you already set your encryption_version to 1? since you are on 1.38 (i'm not entirely sure here).

I wouldnt know whats going wrong otherwise.

**Comment by kielczu** [comment]
Yes i did try setting encryption key to one. It appears that during bluetooth addition the keys are changed from default or the default key is different in the new devices

**Comment by RobHofmann** [comment]
Yes this is correct. During setting up the HVAC, the encryptionkey will be set. So after setting up the HVAC to your WiFi, you should make sure this encryption key is matching the new version.

Fastest way (for me at least) is to simply let the integration handle this. So i'm NOT setting the encryption key manually in my config.

Can you share the logs with encryption_version set to 1 and no manual encryption key set? I'm curious if it still will say "Error getting encryptoin key".

**Comment by RobHofmann** [comment]
> I have the same issue. My device type is: GWH12AGC-K6DNA1A

Please also share your config :)

**Comment by RobHofmann** [comment]
> Hi, I've changed to encryption_version: 2, and now it's working fine.


Glad to hear :)!

**Comment by philipposkhos** [comment]
I have a same issue. I got "2025-06-21 12:05:16.882 INFO (SyncWorker_2) [custom_components.gree.climate] Error getting device encryption key!" wrong on all of my 3 devices. (tried version 1 or 2)

I tried to get the encryption key from my phone, but I found that my app's package name is gree.greeplus and it seems unable to back up the app data, as I only got a 1KB backup.db file after running the adb backup command.

My AC's wifi module version is :" V1.31

Thanks

**Comment by avm99963** [comment]
I'm experiencing the same using a XE7A-24/HC which had its Wi-Fi set up via Bluetooth with the EWPE Smart app. I've attempted to use both encryption version 1 and 2 with the same error ("Error getting device encryption key!"):

```
2025-09-01 01:52:50.793 INFO (MainThread) [custom_components.gree.climate] Initialize the GREE climate device
2025-09-01 01:52:50.793 INFO (MainThread) [custom_components.gree.climate] Unit of measurement: °C
2025-09-01 01:52:50.793 DEBUG (MainThread) [custom_components.gree] Setting up config entry 01K418RGV99Z5CHTGGSKK7Y9JM with data: {'name': 'Name', 'host': '192.168.34.108', 'mac': 'C0:39:37:A6:F7:25', 'port': 7000, 'timeout': 10, 'encryption_version': 2}
2025-09-01 01:52:50.793 DEBUG (MainThread) [custom_components.gree.climate] name(): Name
2025-09-01 01:52:50.793 DEBUG (MainThread) [custom_components.gree.climate] name(): Name
2025-09-01 01:52:50.793 DEBUG (MainThread) [custom_components.gree.climate] supported_features(): 937
2025-09-01 01:52:50.793 DEBUG (MainThread) [custom_components.gree.climate] temperature_unit(): °C
2025-09-01 01:52:50.793 DEBUG (MainThread) [custom_components.gree.climate] hvac_modes(): [<HVACMode.AUTO: 'auto'>, <HVACMode.COOL: 'cool'>, <HVACMode.DRY: 'dry'>, <HVACMode.FAN_ONLY: 'fan_only'>, <HVACMode.HEAT: 'heat'>, <HVACMode.OFF: 'off'>]
2025-09-01 01:52:50.793 DEBUG (MainThread) [custom_components.gree.climate] min_temp(): 16
2025-09-01 01:52:50.793 DEBUG (MainThread) [custom_components.gree.climate] max_temp(): 30
2025-09-01 01:52:50.793 DEBUG (MainThread) [custom_components.gree.climate] target_temperature_step(): 1
2025-09-01 01:52:50.793 DEBUG (MainThread) [custom_components.gree.climate] fan_list(): ['auto', 'low', 'medium_low', 'medium', 'medium_high', 'high', 'turbo', 'quiet']
2025-09-01 01:52:50.793 DEBUG (MainThread) [custom_components.gree.climate] swing_modes(): ['default', 'swing_full', 'fixed_upmost', 'fixed_middle_up', 'fixed_middle', 'fixed_middle_low', 'fixed_lowest', 'swing_downmost', 'swing_middle_low', 'swing_middle', 'swing_middle_up', 'swing_upmost']
2025-09-01 01:52:50.793 DEBUG (MainThread) [custom_components.gree.climate] swing_horizontal_modes(): ['default', 'swing_full', 'fixed_leftmost', 'fixed_middle_left', 'fixed_middle', 'fixed_middle_right', 'fixed_rightmost']
2025-09-01 01:52:50.793 DEBUG (MainThread) [custom_components.gree.climate] supported_features(): 937
2025-09-01 01:52:50.793 INFO (MainThread) [custom_components.gree.climate] Gree climate device added to hass()
2025-09-01 01:52:50.793 DEBUG (MainThread) [custom_components.gree.climate] update()
2025-09-01 01:52:50.793 INFO (MainThread) [custom_components.gree.gree_protocol] Retrieving HVAC encryption key (GCM)
2025-09-01 01:52:50.794 DEBUG (MainThread) [custom_components.gree.gree_protocol] Fetching(192.168.34.108, 7000, 10, {"cid": "app","i": 1,"pack": "Jtod3XIt89Kwts+OaX4UAcl2utYAzgfkbQU3H+DJxfjCLz5YjFc4fdqg9aCGHseEwf35XMKtIPQDq97Iev6J","t":"pack","tcid":"c03937a6f725","uid": 0, "tag" : "MgI+uuNU9C0I/ApUJD8nXg=="})
2025-09-01 01:53:00.804 INFO (MainThread) [custom_components.gree.gree_protocol] Error getting device encryption key!
2025-09-01 01:53:00.805 DEBUG (MainThread) [custom_components.gree.climate] supported_features(): 937
2025-09-01 01:53:00.805 DEBUG (MainThread) [custom_components.gree.climate] temperature_unit(): °C
2025-09-01 01:53:00.805 DEBUG (MainThread) [custom_components.gree.climate] hvac_modes(): [<HVACMode.AUTO: 'auto'>, <HVACMode.COOL: 'cool'>, <HVACMode.DRY: 'dry'>, <HVACMode.FAN_ONLY: 'fan_only'>, <HVACMode.HEAT: 'heat'>, <HVACMode.OFF: 'off'>]
2025-09-01 01:53:00.805 DEBUG (MainThread) [custom_components.gree.climate] min_temp(): 16
2025-09-01 01:53:00.805 DEBUG (MainThread) [custom_components.gree.climate] max_temp(): 30
2025-09-01 01:53:00.805 DEBUG (MainThread) [custom_components.gree.climate] target_temperature_step(): 1
2025-09-01 01:53:00.805 DEBUG (MainThread) [custom_components.gree.climate] fan_list(): ['auto', 'low', 'medium_low', 'medium', 'medium_high', 'high', 'turbo', 'quiet']
2025-09-01 01:53:00.805 DEBUG (MainThread) [custom_components.gree.climate] swing_modes(): ['default', 'swing_full', 'fixed_upmost', 'fixed_middle_up', 'fixed_middle', 'fixed_middle_low', 'fixed_lowest', 'swing_downmost', 'swing_middle_low', 'swing_middle', 'swing_middle_up', 'swing_upmost']
2025-09-01 01:53:00.805 DEBUG (MainThread) [custom_components.gree.climate] swing_horizontal_modes(): ['default', 'swing_full', 'fixed_leftmost', 'fixed_middle_left', 'fixed_middle', 'fixed_middle_right', 'fixed_rightmost']
2025-09-01 01:53:00.805 INFO (MainThread) [custom_components.gree.climate] available(): Device is offline
2025-09-01 01:53:00.805 DEBUG (MainThread) [custom_components.gree.climate] name(): Name
2025-09-01 01:53:00.805 DEBUG (MainThread) [custom_components.gree.climate] supported_features(): 937
2025-09-01 01:53:00.805 DEBUG (MainThread) [custom_components.gree.climate] should_poll()
2025-09-01 01:53:00.808 DEBUG (MainThread) [custom_components.gree.select] Restored select.name_external_temperature_sensor state: None
```

Thanks in advance!

**Comment by RobHofmann** [comment]
Please try again with latest version.

Closing for now to declutter stale issues. Feel free to re-open.
